lib/variable/include/scipp/variable/transform.h

```diff
@@ -100,6 +100,10 @@ inline constexpr auto stride_special_cases<3, false> =
     std::array<std::array<scipp::index, 3>, 3>{
         {{1, 1, 1}, {1, 0, 1}, {1, 1, 0}}};
 
+template <>
+inline constexpr auto stride_special_cases<5, false> =
+    std::array<std::array<scipp::index, 5>, 1>{{{1, 0, 0, 1, 0}}};
```

Comment:
In the example I tested this gave nearly 20% speedup. Should we add more like this, or remove?

Reply:
I tried for a very simple case, `fmadd` (a*b + c). Adding a stride special case gives 5% speedup.

Reply:
How many 4-ary kernels are there? That is, how often do we hit this case and generate code for it? If it's just for user defined kernels, then I don't see an issue with having the special cases.

Reply:
I think there are none used internally, i.e., only user-defined (currently).

Reply:
I am adding more cases then. To limit the number for 5 inputs, I moved all the zeros to the end. This lets use achieve optimal performance by reordering inputs in our own code, I think?